.github/workflows/gradle.yml

```
@@ -0,0 +1,34 @@
+# This workflow will build a Kotlin project with Gradle and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Kotlin CI with Gradle
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up JDK 11
+      uses: actions/setup-java@v3
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+    - name: Build with Gradle
+      uses: gradle/gradle-build-action@v2
+      with:
+        arguments: build
+    - name: Test with Gradle
+      uses: gradle/gradle-build-action@v2
+      with:
+        arguments: test
```

Build step already performs tests.

```suggestion
    - name: Test with Gradle
      uses: gradle/gradle-build-action@v2
      with:
        arguments: test
```